src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -0,0 +1,308 @@
+    return ContainerUtil.groupBy(fieldReferenceExpressions, GoMoveToStructInitializationIntention::getIdentifierName).entrySet().stream()
+      .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().size()));
+  }
+
+  @NotNull
+  private static String getIdentifierName(@NotNull GoReferenceExpression fieldReferenceExpression) {
+    return fieldReferenceExpression.getIdentifier().getText();
+  }
+
+  @NotNull
+  private static Collection<? extends PsiElement> getLeftHandElements(@NotNull GoStatement statement) {
+    if (statement instanceof GoSimpleStatement) {
+      GoShortVarDeclaration varDeclaration = ((GoSimpleStatement)statement).getShortVarDeclaration();
+      return varDeclaration != null ? varDeclaration.getVarDefinitionList() : ContainerUtil.emptyList();
+    }
+    if (statement instanceof GoAssignmentStatement) {
+      return ((GoAssignmentStatement)statement).getLeftHandExprList().getExpressionList();
+    }
+    return ContainerUtil.emptyList();
+  }
+
+  @Override
+  public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element) throws IncorrectOperationException {
+    Pair<List<GoReferenceExpression>, GoCompositeLit> pair = getFieldReferencesAndStructLiteral(element);
+    if (pair.first == null || pair.first.isEmpty()) return;
+    moveFieldReferenceExpressions(pair.first, pair.second, project);
+  }
+
+  private static void moveFieldReferenceExpressions(@NotNull List<GoReferenceExpression> fieldReferenceExpressions,
+                                                    @NotNull GoCompositeLit structLiteral,
+                                                    @NotNull Project project) {
+    GoAssignmentStatement assignment = getValidAssignmentParent(ContainerUtil.getFirstItem(fieldReferenceExpressions));
+    if (assignment == null) return;
+
+    GoLiteralValue literalValue = structLiteral.getLiteralValue();
+    if (literalValue == null) return;
+
+    for (GoExpression expression : fieldReferenceExpressions) {
+      GoReferenceExpression fieldReferenceExpression = tryGetReferenceExpression(expression);
+      if (!isFieldReferenceExpression(fieldReferenceExpression)) continue;
+
+      GoExpression fieldValue = GoPsiImplUtil.getRightExpression(fieldReferenceExpression, assignment);
+      if (fieldValue == null) continue;
+
+      GoPsiImplUtil.deleteExpressionFromAssignment(assignment, fieldReferenceExpression);
+      addFieldDefinition(literalValue, getIdentifierName(fieldReferenceExpression), fieldValue.getText(), project);
+    }
+  }
+
+  private static void addFieldDefinition(@NotNull GoLiteralValue literalValue,
+                                         @NotNull String name,
+                                         @NotNull String value,
+                                         @NotNull Project project) {
+    PsiElement newField = GoElementFactory.createNamedStructField(project, name, value);
```

Comment:
bad name and signature: better `createLiteralValueElement(String key, String value)`